docs/tutorials/security/part-2.md

```diff
@@ -0,0 +1,266 @@
+Copy the contract address, in my case KT1CHJgXEdBPktNNPGTDaL8XEAzJV9fjSkrZ, and paste it on the file `3-reentrancyMaliciousContract.storageList.jsligo` with your value
+
+Compile/deploy the last contract, the Malicious contract which will loop and steal the funds of the Offer contract
+
+```bash
+taq compile 3-reentrancyMaliciousContract.jsligo
+taq deploy 3-reentrancyMaliciousContract.tz -e testing
+```
+
+Copy the contract address, in my case KT1NKLZE9HkGJxjopowLqxA4pswutgMrrXyE, and initialize the Ledger contract as the Malicious contract as some cookies on its storage. Paste the value in the file `3-reentrancyLedgerContract.parameterList.jsligo`
+
+Once done, compile the Ledger contracts and call with this parameter
+
+```bash
+taq compile 3-reentrancyLedgerContract.jsligo
+taq call 3-reentrancyLedgerContract --param 3-reentrancyLedgerContract.parameter.default_parameter.tz -e testing
+```
+
+Context is ready :
+
+- the Malicious contract has cookies on the Ledger contract
+- all deployed contract points to the correct addresses
+
+Now the Malicious contract will try to steal funds from the Offer contract, run the command to start the attack the transaction flow
+
+```bash
+octez-client transfer 0 from alice to KT1NKLZE9HkGJxjopowLqxA4pswutgMrrXyE --entrypoint attack --arg 'Unit' --burn-cap 1
+```
+
+Here you can see the result on the Ghostnet: https://ghostnet.tzkt.io/KT1NKLZE9HkGJxjopowLqxA4pswutgMrrXyE/operations/
+
+3 refunds will be emitted instead of one
+
+&rarr; **SOLUTION**: on the `3-reentrancyOfferContract.jsligo` file, line 34, swap the order of operation execution
+
+from
+
+```
+return [list([opTx, opChangeOwner]), s];
+```
+
+to
+
+```
+return [list([opChangeOwner,opTx]), s];
+```
+
+and rerun the scenario from scratch redeploying the contracts. It should be impossible to run the attack, as the transaction will fail
+
+```logs
+"message":"user do not have cookies"
+```
+
+- Authorize withdraw transfer only to a user account: As User wallet cannot do callback loops, it solves the issue but this solution is not always feasible and limiting. To check if an address is implicit, the Tezos.get_sender and the Tezos.get_source are always equal.
+
+- Audit External Contract calls: This is very hard to check, for example on withdrawal for a token transfer, any contract can receive funds.
+
+- Call third-party security experts or employ automated security tools: If you are not sure about your code, they will identify weaknesses and validate the contract’s security measures.
+
+## Overflow
+
+Manipulating arithmetic operations can lead to overflows and underflows
+
+- On Solidity: SafeMath is a library in Solidity that was designed to provide safe mathematical operations. It prevents overflow and underflow errors when working with unsigned integers (uint), which can lead to unexpected behavior in smart contracts. However, since Solidity v0.8.0, this library has been made obsolete as the language itself starts to include internal checking.
```

Comment:
Is this relevant in a Tezos-focused tutorial?

Reply:
yes, it is to compare both chains implementation differencies